File: 백준/Silver/10773. 제로/제로.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int size = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();
        stack.push(Integer.parseInt(br.readLine()));

        for (int i = 0; i < size - 1; i++) {
            int n = Integer.parseInt(br.readLine());
            if (n == 0) stack.pop();
            else stack.push(n);
        }
        int sum = 0;
        for (Integer i : stack) {
            sum += i;
        }
        System.out.println(sum);
    }
}

File: 백준/Silver/10773. 제로/제로.kt
import java.io.*
import java.util.*

fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    var stack = Stack<Int>()
    repeat(br.readLine().toInt()) {
        val num = br.readLine().toInt()
        if(num == 0) {
            stack.pop()
        } else {
            stack.push(num)
        }
    }
    val sum = stack.sum()
    bw.write(sum.toString())
    bw.flush()
    bw.close()
}